File: python/exercise_290.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        s = s.split()
        if len(s) != len(pattern):
            return False
        hashmap = {}
        hashmap2 = {}
        for i in range(len(pattern)):
            if pattern[i] not in hashmap:
                hashmap[pattern[i]] = s[i]               
            if s[i] not in hashmap2:
                hashmap2[s[i]] = pattern[i]
        for i in range(len(pattern)):
            if not(hashmap[pattern[i]] == s[i] and hashmap2[s[i]] == pattern[i]):
                return False
        return True

File: java/hashsetOrHashmap/exercise_290.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        char[] patternLetters = pattern.toCharArray();
        String[] sWords = s.split(" ");
        if (sWords.length!=patternLetters.length){
            return false;
        }

        HashMap<String,Character> a = new HashMap<>();
        HashMap<Character,String> b = new HashMap<>();
        for(int i=0;i<sWords.length;i++){
            if(a.containsKey(sWords[i])){
                if(a.get(sWords[i])!=patternLetters[i]){
                    return false;
                }
            }
            a.put(sWords[i],patternLetters[i]);

            if(b.containsKey(patternLetters[i])){
                if(!b.get(patternLetters[i]).equals(sWords[i])){
                    return false;
                }
            }
            b.put(patternLetters[i],sWords[i]);
        }
        return true;
    }
}